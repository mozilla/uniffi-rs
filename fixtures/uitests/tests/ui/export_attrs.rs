fn main() {} /* empty main required by `trybuild` */

#[uniffi::constructor] // <--- should be an error!
// Someone might try to 'export' a struct instead of deriving it.
#[uniffi::export]
struct S {}

#[uniffi::export(Dixplay)]
struct S2 {}

#[derive(uniffi::Object)]
struct Object;

#[uniffi::export(callback_interface)]
impl Object {}

#[uniffi::export(with_foreign)]
fn foreign() {}

uniffi_macros::setup_scaffolding!();
