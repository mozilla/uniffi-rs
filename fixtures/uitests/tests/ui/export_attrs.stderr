error: uniffi::export on a struct must supply a builtin trait name. Did you mean `#[derive(uniffi::Object)]`?
 --> tests/ui/export_attrs.rs:5:1
  |
5 | #[uniffi::export]
  | ^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `uniffi::export` (in Nightly builds, run with -Z macro-backtrace for more info)

error: uniffi::export struct attributes must be builtin trait names; `Dixplay` is invalid
 --> tests/ui/export_attrs.rs:8:18
  |
8 | #[uniffi::export(Dixplay)]
  |                  ^^^^^^^

error: uniffi::export attribute `callback_interface` is not supported here.
  --> tests/ui/export_attrs.rs:14:18
   |
14 | #[uniffi::export(callback_interface)]
   |                  ^^^^^^^^^^^^^^^^^^

error: uniffi::export attribute `with_foreign` is not supported here.
  --> tests/ui/export_attrs.rs:17:18
   |
17 | #[uniffi::export(with_foreign)]
   |                  ^^^^^^^^^^^^
